src/dispatch/conversation/flows.py

```diff
@@ -44,6 +46,13 @@ def create_case_conversation(
 
     conversation = None
 
+    # Do not overwrite a case conversation with one of the same type (thread, channel)
+    if case.conversation:
+        if case.has_channel:
+            return case.conversation
+        if case.has_thread and not case.dedicated_channel:
+            return case.conversation
```

Comment:
suggestion: maybe we could raise an exception here if usage of `create_case_conversation` when one of these cases exists, i worry it may be unintuitive for the failure state to return the existing conversation and/or introduce some other potential bugs downstream

Reply:
Done.